22.2.5+ (#1)
friedcucumber/maui-data-grid-search-bar: README.md

<!-- default badges list -->
<!-- default badges end -->
# DataGrid for .NET MAUI - Search Bar

> This example shows how to implement a [Data Grid](https://docs.devexpress.com/MAUI/403255/data-grid/data-grid) search. The search bar is placed into the page's [toolbar area](https://learn.microsoft.com/en-us/dotnet/api/microsoft.maui.controls.toolbaritem).

Reviewer: This introductory paragraph has exactly the same issue as its documentation counterpart. Please refer to the corresponding comment on GitLab for my reasoning.

```suggestion
This example demonstrates how to place a search bar in the [toolbar area](https://learn.microsoft.com/en-us/dotnet/api/microsoft.maui.controls.toolbaritem) of [Data Grid](https://docs.devexpress.com/MAUI/403255/data-grid/data-grid) view.
```

Author: Bringing here the sentence from our GitLab discussions:

```suggestion
This example demonstrates how to place a [Data Grid](https://docs.devexpress.com/MAUI/403255/data-grid/data-grid) search bar in a [ContentPage](https://learn.microsoft.com/en-us/dotnet/maui/user-interface/pages/contentpage)'s [toolbar area](https://learn.microsoft.com/en-us/dotnet/api/microsoft.maui.controls.toolbaritem).
```

Reviewer: Yes, everything is correct. Thank you!

<img width="30%" src="https://user-images.githubusercontent.com/12169834/230120654-1b2536ff-757a-499b-873b-5eb5d13b47ad.png"/>

## Implementation Details

> * The [TextEdit](https://docs.devexpress.com/MAUI/DevExpress.Maui.Editors.TextEdit) control is placed in the title bar. You can specify the [EndIcon](https://docs.devexpress.com/MAUI/DevExpress.Maui.Editors.EditBase.ErrorIcon) property to define the right icon. 

Reviewer 2: "the right icon" could mean either "the icon on the right" or "the correct icon". Instead, can we change it to something like "...to set the error icon"?

Author: ```suggestion
* The [TextEdit](https://docs.devexpress.com/MAUI/DevExpress.Maui.Editors.TextEdit) control is placed in the title bar. You can specify the [EndIcon](https://docs.devexpress.com/MAUI/DevExpress.Maui.Editors.EditBase.EndIcon) property to define the icon on the right. 
```

* Specify the [PlaceholderText](https://docs.devexpress.com/MAUI/DevExpress.Maui.Editors.EditBase.PlaceholderText) property to define TextEdit's placeholder text. Define the [PlaceholderColor](https://docs.devexpress.com/MAUI/DevExpress.Maui.Editors.EditBase.PlaceholderColor) property to specify the placeholder text.
* You can use the [ClearIconColor](https://docs.devexpress.com/MAUI/DevExpress.Maui.Editors.EditBase.ClearIconColor) property to define a color of TextEdit's clear icon.

> * On the [TextChanged](https://docs.devexpress.com/MAUI/DevExpress.Maui.Editors.TextEditBase.TextChanged) event, you can define a query parameter and pass it to the [DataGridView.FilterString](https://docs.devexpress.com/MAUI/DevExpress.Maui.DataGrid.DataGridView.FilterString). Refer to the following topic for more information on the filter criteria syntax: [Criteria Language Syntax](https://docs.devexpress.com/CoreLibraries/4928/devexpress-data-library/criteria-language-syntax).

Reviewer: > On the `TextChanged` event?

Did you want to say "In a TextChanged event handler"? That would have been much more technically accurate in my book.

> you can define a query parameter and **pass it** to the [DataGridView.FilterString](https://docs.devexpress.com/MAUI/DevExpress.Maui.DataGrid.DataGridView.FilterString)

Well, `FilterString` is a property, therefore, you can assign your query parameter to it rather than pass. Normally, I would have expected the verb "pass" if you passed a value as a method parameter. Or passed it to a method. Or into an event handler. In this case, everything should be much simpler in technical sense.

```suggestion
* In a [TextChanged](https://docs.devexpress.com/MAUI/DevExpress.Maui.Editors.TextEditBase.TextChanged) event handler, you can define a query parameter and assign it to the [DataGridView.FilterString](https://docs.devexpress.com/MAUI/DevExpress.Maui.DataGrid.DataGridView.FilterString). Refer to the following topic for more information on the filter criteria syntax: [Criteria Language Syntax](https://docs.devexpress.com/CoreLibraries/4928/devexpress-data-library/criteria-language-syntax).
```
